api/client/events.go

```diff
@@ -31,6 +31,13 @@ func EventToGRPC(in types.Event) (*proto.Event, error) {
 		Type: eventType,
 	}
 	if in.Type == types.OpInit {
+		watchStatus, ok := in.Resource.(*types.WatchStatusV1)
```

Comment:
I decided not to do that because `EventToGRPC` only works with events originated from the current version of code which attaches `WatchStatusV1` to `OpInit` events everywhere. I inspected all usages of `OpInit` to make sure of that.

Let me know if you still have concerns about this.

Reply:
Fair enough - the error is gonna be enough to tip us off if we make a mistake in that regard.